config/FreeRTOSConfig.h

```diff
@@ -114,8 +114,8 @@ to exclude the API function. */
 #define vPortSVCHandler SVC_Handler
 
 //Milliseconds to OS Ticks
-#define M2T(X) ((unsigned int)(X*(configTICK_RATE_HZ/1000.0)))
-#define F2T(X) ((unsigned int)((configTICK_RATE_HZ/X)))
+#define M2T(X) ((unsigned int)((X)*(configTICK_RATE_HZ/1000.0)))

```

Comment:
Interestingly, the optimizer falls over if you replace `1000.0` here with `1000.0f`.  I haven't investigated further.